core/src/main/java/org/apache/carbondata/core/metadata/datatype/DecimalConverterFactory.java

```diff
@@ -261,18 +266,22 @@ void fillVector(Object valuesToBeConverted, int size, ColumnVectorInfo info, Bit
       return new BigDecimal(bigInteger, scale);
     }
 
-    @Override public void fillVector(Object valuesToBeConverted, int size, ColumnVectorInfo info,
-        BitSet nullBitset, DataType pageType) {
-      CarbonColumnVector vector = info.vector;
-      int precision = info.measure.getMeasure().getPrecision();
-      int newMeasureScale = info.measure.getMeasure().getScale();
+    @Override public void fillVector(Object valuesToBeConverted, int size,
+        ColumnVectorInfo vectorInfo, BitSet nullBitSet, DataType pageType) {
+      CarbonColumnVector vector = vectorInfo.vector;
```

Comment:
Can you refactor this code, move to some method

Reply:
done